torch/optim/lr_scheduler.py

```diff
@@ -1761,6 +1761,18 @@ def _format_param(self, name, optimizer, param):
         else:
             return [param] * len(optimizer.param_groups)
 
+
+    def state_dict(self):
```

Comment:
Use versioning here to ensure old `state_dict` can be loaded

Reply:
I think a better refactor is:
- Add a `self._anneal_func_type` that is the string from the user
- Add a `self._anneal_func` method on the class that will check the _anneal_func_type field and call the appropriate method or will call self.anneal_func when that exists (for BC)

That way you don't need to override state_dict/load_state_dict

Reply:
Made the suggested fix